--- a/src/openvpn/forward.c
+++ b/src/openvpn/forward.c
@@ -943,7 +943,10 @@ read_incoming_link(struct context *c, st
 
     status = link_socket_read(sock,
                               &c->c2.buf,
-                              &c->c2.from);
+                              &c->c2.from,
+                              c->options.ce.xormethod,
+                              c->options.ce.xormask,
+                              c->options.ce.xormasklen);
 
     if (socket_connection_reset(sock, status))
     {
@@ -1803,7 +1806,10 @@ process_outgoing_link(struct context *c,
                 /* Send packet */
                 size = (int)link_socket_write(sock,
                                               &c->c2.to_link,
-                                              to_addr);
+                                              to_addr,
+                                              c->options.ce.xormethod,
+                                              c->options.ce.xormask,
+                                              c->options.ce.xormasklen);
 
                 /* Undo effect of prepend */
                 link_socket_write_post_size_adjust(&size, size_delta, &c->c2.to_link);
